pkg/ccl/streamingccl/streamproducer/event_stream.go

```diff
@@ -340,6 +348,37 @@ func (s *eventStream) streamLoop(ctx context.Context, frontier *span.Frontier) e
 		batchSize += keyValue.Size()
 	}
 
+	scanSST := func(sst *roachpb.RangeFeedSSTable) error {
+		iter, err := storage.NewMemSSTIterator(sst.Data, true)
+		if err != nil {
+			return err
+		}
+		for _, sp := range s.spec.Spans {
+			iter.SeekGE(storage.MVCCKey{Key: sp.Key})
+			endKey := storage.MVCCKey{Key: sp.EndKey}
+			for ; ; iter.Next() {
+				if ok, err := iter.Valid(); err != nil {
+					return err
+				} else if !ok {
+					break
+				} else if !iter.UnsafeKey().Less(endKey) {
+					break
+				}
+
+				keyValue := roachpb.KeyValue{
+					Key: iter.UnsafeKey().Key,
+					Value: roachpb.Value{
+						RawBytes:  iter.UnsafeValue(),
+						Timestamp: iter.UnsafeKey().Timestamp,
+					},
+				}
+				batch.KeyValues = append(batch.KeyValues, keyValue)
```

Comment:
Doesn't it defeat the purpose of ssts -- i.e. exploding them into key/values? That's already happening btw; If you need to trim things, perhaps you should either regenerate sstable... Or... perhaps do it on the receiving end (it's true -- there is more data sent, but the sender usually needs to rekey anyway -- so will be iterating sst anyway.

Maybe there is another way of doing this altogether -- but sending 1 key at a time is not it.

Reply:
+1. The target cluster should ingest these SSTs directly, similarly to the source cluster. Otherwise, if the source is running a large import and the target cluster ingests these as regular point writes, the target cluster won't be able to keep up at all.

Reply:
Which reminds me: we may want to use a similar optimization for initial/catchup scans during streaming replication, such that this data is bulk loaded rather than ingested as point writes.

Reply:
Hmm.. Yeah, I guess we really should emit initial scan as ssts.  Probably something that should go into rangefeed library.